packages/editor/src/components/provider/index.js

```diff
@@ -222,6 +223,37 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 			setEditedPost,
 			setRenderingMode,
 		} = unlock( useDispatch( editorStore ) );
+
+		const { computedSectionRootClientId } = useSelect(
+			( select ) => {
+				const { getBlockAttributes, getBlocksByName } =
+					select( blockEditorStore );
+
+				let _sectionRootClientId;
+
+				if ( mode === 'template-locked' ) {
+					_sectionRootClientId =
+						getBlocksByName( 'core/post-content' )?.[ 0 ] ?? '';
+				} else {
+					_sectionRootClientId =
+						getBlocksByName( 'core/group' ).find(
+							( clientId ) =>
+								getBlockAttributes( clientId )?.tagName ===
+								'main'
+						) ?? '';
+				}
+
+				return {
+					computedSectionRootClientId: _sectionRootClientId,
+				};
+			},
+			[ mode ]
+		);
```

Comment:
Essentially this is the algorithm that determines the section "root" based on things that are specific to WordPress. For example the `core/post-content` block is likely WP-specific and so we can't embed this logic in the block editor package.

We could move this into a hook to better name the concept and make the code easier to reason about.

What I'm really interested to agree on is whether this is the best _location_ to be doing such setup logic?